build/Dockerfile

```diff
@@ -8,7 +8,12 @@ COPY go.mod go.sum ./
 COPY collector ./collector
 COPY client ./client
 RUN GO111MODULE=on CGO_ENABLED=0 GOOS=linux go build -mod=vendor -a -installsuffix cgo -ldflags "-X main.version=${VERSION} -X main.gitCommit=${GIT_COMMIT}" -o exporter .
+RUN useradd -u 1001 nginx-prometheus-exporter
 
-FROM alpine:3.11
+FROM scratch
 COPY --from=builder /go/src/github.com/nginxinc/nginx-prometheus-exporter/exporter /usr/bin/
+COPY --from=builder /etc/passwd /etc/passwd
```

Comment:
is copying `/etc/passwd` necessary?
if we don't specify it, `USER 1001` will still work. The only difference I see is that group id will be `0`, which we can change with `USER 1001:1001`

Reply:
@alex1989hu
Since it's not needed and doesn't cause issues for the exporter, could you remove `/etc/passwd` file and use `USER 1001:1001` instead? We'd be happy to merge the PR once the file is removed.

Thanks for providing additional rational here https://github.com/nginxinc/nginx-prometheus-exporter/issues/102#issuecomment-642128681

Reply:
Hello @pleshakov, I made some investigation, I found that `nsswitch.conf` shall be also part of the image to avoid possible hostname/dns resolution issue. I would change my mind to use `gcr.io/distroless/static:nonroot` which content can be found here (`nobody` user is not mention, but part of `nonroot` variant: https://github.com/GoogleContainerTools/distroless/tree/master/base#image-contents

Also, kubernetes related images already started to use it. What do you think, can we choose this way?

Reply:
Hi @alex1989hu

Thanks for further investigating potential problems!

The mentioned distroless image also includes CA certs so that there is no need to copy them from the go image.

Let me further discuss it with the team and get back to you next week.

Reply:
Hi @alex1989hu 

Let's go ahead with image `gcr.io/distroless/static` image. However, could you use `USER 1001:1001` in the Dockerfile?

Reply:
Hello @pleshakov, what about using `gcr.io/distroless/static:nonroot` variant? Pros: it has the user and don't need to use non-existing `uid`:`gid`.  We can use `nobody` - the `nonroot` has a home directory but this is not needed for `nginx-prometheus-exporter`. Thoughts?

Reply:
Because the exporter doesn't need a homedirectory and doesn't depend on `/etc/passwd`, I'd rather have us controlling which user we use in the Dockerfile and defining that explicitly with the `USER`.
Also, the image `static:nonroot` doesn't seem to be well documented in https://github.com/GoogleContainerTools/distroless/

Reply:
👍 => updated description and applied requested change